friktion/friktion_swap_client/swap.py

```diff
@@ -0,0 +1,521 @@
+
+    """
+    Object used to interact with swap contract
+    Args:
+        config (ContractConfig): Configuration to setup the Contract
+    """
+
+    async def get_otoken_details_for_offer(self, offer: Offer):
+        if offer.swapOrderAddress is None:
+            raise Exception('can only get otoken details on an offer associated with an existin swap order')
+        print('swap order address = ', offer.swapOrderAddress)
+        swap_order = await self.get_swap_order_for_key(offer.swapOrderAddress)
+        options_contract = await self.get_options_contract_for_key(swap_order.options_contract)
+        ul_factor = await self._get_token_norm_factor(options_contract.underlying_mint)
+        quote_factor = await self._get_token_norm_factor(options_contract.quote_mint)
+        strike_price = (ul_factor / quote_factor) * options_contract.quote_amount / options_contract.underlying_amount if options_contract.is_call else (quote_factor / ul_factor) * options_contract.underlying_amount / options_contract.quote_amount
+        return {
+            'underlyingAsset': options_contract.underlying_mint,
+            'expiryTimestamp': options_contract.expiry_ts,
+            'isPut': not options_contract.is_call,
+            'strikeAsset': options_contract.quote_mint,
+            'strikePrice': strike_price,
+            'collateralAsset': options_contract.underlying_mint
+        }
+
+    async def _get_token_norm_factor(self, mint: PublicKey) -> int:
+        client = AsyncClient(self.url)
+        await client.is_connected()
+        
+        give_token = AsyncToken(
+            client,
+            mint,
+            TOKEN_PROGRAM_ID,
+            Keypair.generate()
+        )
+
+        return 10 ** (await give_token.get_mint_info()).decimals
+
+    async def get_options_contract_for_key(self, key: PublicKey) -> OptionsContract:
+        client = AsyncClient(self.url)
+        res = await client.is_connected()
+        
+        acc = await OptionsContract.fetch(client, key)
+
+        if acc is None:
+            raise ValueError("No options contract found for key = ", key)
+
+        await client.close()
+        return acc
+
+    async def get_swap_order_for_key(self, key: PublicKey) -> SwapOrder:
+        client = AsyncClient(self.url)
+        res = await client.is_connected()
+        
+        acc = await SwapOrder.fetch(client, key)
+
+        if acc is None:
+            raise ValueError("No swap order found for key = ", key)
+            
+        await client.close()
+        return acc
+
+    async def get_swap_order(self, user: PublicKey, order_id: int) -> SwapOrder:
+        [addr, bump] = find_swap_order_address(user, order_id)
+        
+        acc = await self.get_swap_order_for_key(addr)
+        if acc is None:
+            raise ValueError("No swap found for user = ", user, ', order id = ', order_id)
+
+        return acc
+
+
+    async def get_offer_details(self, user: PublicKey, order_id: int) -> Offer:
```

Comment:
I see the user is used only to fetch the `swap_order_address`...

Is the `swap_order_address` what you are going to provide as the `verifying_contract` at rfq creation?
If so, we could simplify this method to take that as the argument, not requiring the user anymore.

Reply:
yeah, we can simplify this - but the authority for this hasn't been creatd yet so we don't currently know the value - can we change this in the necessary future PR for encryption? should just be a small one